### PR TITLE
Some bugs fixed, better IE support, events added (update, disable, enable)

### LINKED ISSUE #1: Not working in IE8
Is it possible to make it work in IE 8? Basic functionality at least.

Regards,
Raman

### LINKED ISSUE #3: How to disable nice-select on click event of other check box
Hi I have to disable my dropdown on click of checkbox how to implement the same in nice select

### LINKED ISSUE #7: setting selected does not update on screen
Using:
```javascript
$("#selectorTest").val(value);
```
On a `<select>` element does not update the currently selected option.

Using
```javascript
console.log($("#selectorTest").val())
```
Logs the expected value.

Refreshing the page will update the `<select>` to the set option.

Standard `<select>` elements will update their selected option immediately when using jQuery.val()

### LINKED ISSUE #10: Up/down arrow not rendered correctly in IE9.
Instead of rendered as arrows, it looks like this:

screen shot 2016-02-12 at 9 08 09 pm

BTW, it's an awesome plugin! Light-weight, slick, and works almost anywhere!

Would be great if the IE9 arrow issue can be fixed.

### PR DESCRIPTION
**Pull request by implico**
Bug fixed: select input with no 'class' attribute caused adding 'undefined' class to the dropdown container (caused by commit a353ca6) and added an unnecessary space; bug fixed: use within a 'label' tag caused a double-click effect (dropdown not shown); updated: use of 'prop' function to check disabled state; added events: update (fixes #7), disable, enable (fixes #3); CSS: added vendor prefixes to arrow transform property (old IE, Safari) and dropdown opacity (IE<10) (fixes #1, fixes #10)

Plus unmentioned in the commit "data-display" fix (quotes).

Suggested to describe new events in the docs, if acceptable.

Cheers and thank you for a really `nice` simple plugin.

### CODE REVIEW

**Comment by hernansartorio** [comment]
Wow thanks a lot man, I really appreciate all the work you've put into this!

However I can't accept the pull request as is, because there are a lot of changes in one single commit, and some of those changes need to be reviewed. You should try to make one commit or pull request for each logical change in the code—be it a bug fix, improvement or new feature—so it makes it easier to review and track changes later on.

As for the CSS vendor prefixes, they're not included in the SASS file because I'm using PostCSS Autoprefixer to add them automatically after compiling (I know I need to make this clear somewhere in the Readme).

Good catch on the label bug, I'd merge your solution as is if it were a single pull request.

As for the rest of the changes I'll need some more time to review, but then again it would be a lot better if they were on separate commits.

Thanks again!

**Comment by implico** [comment]
Hi Hernán,

I know it is better to perform fixes/changes as single commits, but all of this took ~2h (initially I just wanted to fix the "undefinded" class and "data-display" bugs, then started to look at the issues) and I wouldn't say these changes are very complex, most of them are just small fixes (except events). I propose to test my forked version, everything looks fine. I am afraid I won't have time to split the commit :(

About vendor prefixes - in the documentation it is recommended to use the SASS styles version. Although it is a standard now to use a kind of autoprefixer, I would leave hardcoded prefixes for crucial properties (arrow and opacity) - there are many people using e.g. Compass mixins to do this.

Regards:)

**Comment by hernansartorio** [comment]
I know they're small changes, but still it's a good practice if there is more than one person pushing code to a repo, and I'd rather keep things organized looking forward. I understand if you don't have the time, I really appreciate you pointing out the bugs and taking the time to collaborate though :)

And good point about the vendor prefixes, I didn't think of that. I think it might be better to configure the Gulp tasks to generate another version of the SASS file including the prefixes.

**Comment by GabrielDumbrava** [comment]
Can you please make this PR work. The ability to select a different value from js/Angular is a lifesaver, not to mention that I don't have to hide the extra empty row with CSS.

Thanks for all the effort to make this nice select.

**Comment by Treverix** [comment]
Hi, what's the status of this? Will it be pulled into the repository or will the fixes be applied directly?

It addresses some problems that we have with nice-select and I just wonder if we should wait a bit more or fork the repository and do the implementation by ourselves?

**Comment by hernansartorio** [comment]
> The ability to select a different value from js/Angular is a lifesaver, not to mention that I don't have to hide the extra empty row with CSS.

@GabrielDumbrava can you specify what functionality you mean with that?

**Comment by hernansartorio** [comment]
> It addresses some problems that we have with nice-select...

@Treverix what are those problems?

Some of the issues addressed in this PR are already solved now, only the methods are missing—which I have started working on and will publish soon.

**Comment by GabrielDumbrava** [comment]
@hernansartorio I was referring to [#7](https://github.com/hernansartorio/jquery-nice-select/issues/7) which I only checked with an Angular generated select, but I've checked again with `$("#selectorTest").val(value);` and it worked. I thought it was not fixed. My bad.

File: js/jquery.nice-select.js [hunk 1]
```diff
@@ -31,9 +32,29 @@
...
             '>' + $(this).text() + '</li>');
         });
+        
+        
+        /*
+           Events
+        */
+        
+        // Updates the dropdown after setting the value manually
+        select.on('update', function() {
```
Comment (ProNotion) [inline js/jquery.nice-select.js]: `update` is not a valid event on select boxes it should be `change`? That said, I had a bit of play with this and found that the change event doesn't get triggered when attempting to update the selected item in code, it requires browser interaction and if you trigger it manually you get stuck in an infinite loop and crashes my browser tab.

**Comment by implico** [comment]
Closing as outdated.

**Pull request closed without merging**